hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/DistributionSetManagementTest.java

```diff
@@ -171,6 +176,84 @@ public void entityQueriesReferringToNotExistingEntitiesThrowsException() {
     }
 
     @Test
+    @Description("Verify that a DistributionSet with with invalid properties cannot be created or updated")
+    @ExpectEvents({ @Expect(type = DistributionSetCreatedEvent.class, count = 1),
+            @Expect(type = SoftwareModuleCreatedEvent.class, count = 3),
+            @Expect(type = DistributionSetUpdatedEvent.class, count = 0) })
+    public void createAndUpdateDistributionSetWithInvalidFields() {
+        final DistributionSet set = testdataFactory.createDistributionSet();
+
+        createAndUpdateDistributionSetWithInvalidDescription(set);
+        createAndUpdateDistributionSetWithInvalidName(set);
+        createAndUpdateDistributionSetWithInvalidVersion(set);
+    }
+
+    @Step
+    private void createAndUpdateDistributionSetWithInvalidDescription(final DistributionSet set) {
+
+        assertThatExceptionOfType(ConstraintViolationException.class)
+                .isThrownBy(() -> distributionSetManagement.createDistributionSet(entityFactory.distributionSet()
+                        .create().name("a").version("a").description(RandomStringUtils.randomAlphanumeric(513))))
+                .as("entity with too long description should not be created");
```

Comment:
To make this test more robust you could use NamedEntity.DESCRIPTION_MAX_SIZE + 1 instead assuming 513 as too long.